spec/02-integration/05-proxy/10-balancer/01-ring-balancer_spec.lua

```diff
@@ -1310,6 +1421,77 @@ for _, strategy in helpers.each_strategy() do
             end
           end)
 
+          it("perform active health checks with upstream hostname", function()
+
+            for nfails = 1, 3 do
+
+              local requests = SLOTS * 2 -- go round the balancer twice
+              local port1 = gen_port()
+              local port2 = gen_port()
+
+              -- setup target servers:
+              -- server2 will only respond for part of the test,
+              -- then server1 will take over.
+              local server2_oks = math.floor(requests / 4)
+              local server1 = http_server("localhost", port1,
+                { requests - server2_oks }, "false", "http", "true")
+              local server2 = http_server("localhost", port2, { server2_oks },
+                "false", "http", "true")
+
+              -- configure healthchecks
+              begin_testcase_setup(strategy, bp)
+              local upstream_name, upstream_id = add_upstream(bp, {
+                hostname = "localhost",
```

Comment:
Can you make this a value other than `localhost`, some invalid name like `upstream-hostname.test` to make sure this test does not use the underlying system's name resolution in any way?